src/main/kotlin/skytils/skytilsmod/features/impl/mining/MiningFeatures.kt

```diff
@@ -173,10 +173,66 @@ class MiningFeatures {
         if (Skytils.config.hollowChatCoords && SBInfo.mode == SkyblockIsland.CrystalHollows.mode) {
             val xyzMatcher = xyzPattern.matcher(unformatted)
             val xzMatcher = xzPattern.matcher(unformatted)
-            if (xyzMatcher.matches())
+            if (xyzMatcher.matches()) {
                 waypointChatMessage(xyzMatcher.group("x"), xyzMatcher.group("y"), xyzMatcher.group("z"))
-            else if (xzMatcher.matches())
+                return
+            } else if (xzMatcher.matches()) {
                 waypointChatMessage(xzMatcher.group("x"), "100", xzMatcher.group("z"))
+                return
+            }
+
+            /**
+             * Checks for the format used in DSM and SBE
+             * $DSMCHWP:Mines of Divan@-673,117,426 ✔
+             * $SBECHWP:Khazad-dûm@-292,63,281 ✔
+             * $asdf:Khazad-dûm@-292,63,281 ❌
+             * $SBECHWP:Khazad-dûm@asdf,asdf,asdf ❌
+             */
+            if (unformatted.contains(Regex("\\\$(SBECHWP\\b|DSMCHWP):(.*?)@-?\\d*,-?\\d*,-?\\d*"))) {
+                val sub = unformatted.split(Regex("\\\$(SBECHWP\\b|DSMCHWP):"))[1]
+                val parts = sub.split("@")
+                val location = parts[0]
+                val coords = parts[1].split(",")
+                if (locationsToInternal.containsKey(location)) {
+                    val key = locationsToInternal[location]
+
+                    /**
+                     * Sends the waypoints message except it suggests which one should be used based on
+                     * the name contained in the message and converts it to the internally used names for the waypoints.
+                     */
+                    val component = ChatComponentText(
+                        "§3Skytils > §eFound coordinates in a chat message, click a button to set a waypoint.\n"
+                    )
+                    val suggested = ChatComponentText("§f[${location}] ").setChatStyle(
+                        ChatStyle()
+                            .setChatClickEvent(
+                                ClickEvent(
+                                    ClickEvent.Action.RUN_COMMAND,
+                                    "/skytilshollowwaypoint set $key ${coords[0]} ${coords[1]} ${coords[2]}"
+                                )
+                            ).setChatHoverEvent(
+                                HoverEvent(
+                                    HoverEvent.Action.SHOW_TEXT,
+                                    ChatComponentText("§eset waypoint for $location")
+                                )
+                            )
+                    )
+                    val custom = ChatComponentText("§e[Custom]").setChatStyle(
```

Comment:
this can be simplified

Reply:
Should I remove the declaration of the variable and just append it on creation? I thought it was fine, because it is the same code used later in the file